src/Avalonia.Controls/Control.cs

```diff
@@ -329,5 +347,55 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == FlowDirectionProperty)
+            {
+                InvalidateFlowDirection();
+                
+                foreach (var visual in VisualChildren)
+                {
+                    if (visual is Control child)
+                    {
+                        child.InvalidateFlowDirection();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Computes the <see cref="IVisual.IsMirrorTransform"/> value according to the 
+        /// <see cref="FlowDirection"/> and <see cref="BypassFlowDirectionPolicies"/>
+        /// </summary>
+        public virtual void InvalidateFlowDirection()
```

Comment:
Should it be "protected virtual"? I don't think it's expected to be called outside of the control normally.

Reply:
unless you have another idea for this https://github.com/AvaloniaUI/Avalonia/pull/7810/files/ac18fb35978fe3c6237ce934eb10157c49e237f1#diff-280146b8eb056717d4a98d5ab3d7426a446191339f5dc466a3f0448153bd54dd.